### PR TITLE
Add ARM64 builder images for Golang 1.15

### PR DESCRIPTION
**Pull request by kvch**
### What

Support ARM64 builder image for Golang 1.15 in Debian 9 that are built natively in ARM64 therefore no cross-build images in ARM for ARM.

### Further details

For simplicity we won't build docker images for other Debian versions.

It uses a fixed ARM worker since there are issues with the current docker credentials -> https://github.com/elastic/golang-crossbuild/pull/76#issuecomment-769812513

### Issues

Superseedes https://github.com/elastic/golang-crossbuild/pull/75

### CODE REVIEW

**Comment by v1v** [comment]
### Issues

* For debian8 arm
```
[2021-01-29T11:05:12.248Z] W: GPG error: http://archive.debian.org jessie Release: The following signatures were invalid: KEYEXPIRED 1587841717
```

* For debian9 arm

```
Consumes base instead base-arm

```

**Comment by kvch** [comment]
Well, I haven't seen this error before. o.O Let's see if your updates fix it.

File: Jenkinsfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-@Library('apm@current') _
+@Library('apm@test/docker') _
```
Comment (v1v) [inline Jenkinsfile:2]: For testing purposes only

**Comment by v1v** [comment]
`docker-credential-helpers` seems to be a problematic package, in fact, it's causing some issues for the docker login too:

https://github.com/docker/compose/issues/6023#issuecomment-402067570

File: go1.15/Makefile.common [hunk 2]
```diff
@@ -23,4 +23,20 @@ build:
...
+	--build-arg REPOSITORY=$(REPOSITORY) \
+	--build-arg VERSION=$(VERSION) \
+	--build-arg DEBIAN_VERSION=$(DEBIAN_VERSION) \
+	--build-arg TAG_EXTENSION=$(TAG_EXTENSION) \
+	--build-arg IMAGE="$(REPOSITORY)/$(NAME):$(VERSION)$(SUFFIX)$(TAG_EXTENSION)" \
+	--build-arg VCS_REF="$(VCS_REF)" \
+	--build-arg VCS_URL="$(VCS_URL)" \
+	--build-arg BUILD_DATE="$(BUILD_DATE)" \
+	.
+	docker save -o "$(NAME):$(VERSION)$(SUFFIX)$(TAG_EXTENSION).tar" "$(REPOSITORY)/$(NAME):$(VERSION)$(SUFFIX)$(TAG_EXTENSION)"
```
Comment (v1v) [inline go1.15/Makefile.common:40]:
```suggestion
```

File: Jenkinsfile [hunk 3]
```diff
@@ -72,11 +130,14 @@ pipeline {
             }
           }
           stage('Release') {
+            environment {
+              HOME = "${env.WORKSPACE}"
+            }
```
Comment (v1v) [inline Jenkinsfile:135]:
```suggestion
```

File: Jenkinsfile [hunk 4]
```diff
@@ -47,23 +47,81 @@ pipeline {
...
         }
         stages {
           stage('Build') {
             steps {
-              withGithubNotify(context: "Build ${GO_FOLDER} ${MAKEFILE}") {
+              withGithubNotify(context: "Build ${GO_FOLDER} ${MAKEFILE} ${PLATFORM}") {
                 deleteDir()
                 unstash 'source'
                 buildImages()
+                archiveArtifacts allowEmptyArchive: true, artifacts: '**/*.tar', defaultExcludes: false
```
Comment (v1v) [inline Jenkinsfile:114]:
```suggestion
```

File: Jenkinsfile (hunk 1 above)
Comment (v1v) [reply Jenkinsfile:2]:
```suggestion
@Library('apm@current') _
```

File: Jenkinsfile [hunk 5]
```diff
@@ -61,9 +117,10 @@ pipeline {
           stage('Staging') {
             environment {
               REPOSITORY = "${env.STAGING_IMAGE}"
+              HOME = "${env.WORKSPACE}"
```
Comment (v1v) [inline Jenkinsfile:120]:
```suggestion
```

File: go1.15/base-arm/Dockerfile.tmpl [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+# show the GLIBC version
+RUN ldd --version
+
+WORKDIR /
+RUN go get -d . \
+    && go build -o /crossbuild /entrypoint.go \
+    && rm -rf /go/* /root/.cache/* /entrypoint.go
+
+ENV GOLANG_CROSSBUILD=0
```
Comment (kvch) [inline go1.15/base-arm/Dockerfile.tmpl:65]: This should be set to 1.

Comment (kuisathaverat) [reply go1.15/base-arm/Dockerfile.tmpl:65]: ok, it is a flag on beats https://github.com/elastic/beats/blob/master/dev-tools/mage/build.go#L89

Comment (kuisathaverat) [reply go1.15/base-arm/Dockerfile.tmpl:65]:
```suggestion
ENV GOLANG_CROSSBUILD=1
```

**Review by kvch — COMMENTED** [review]
I approve :D

**Review by kuisathaverat — APPROVED** [review]

**Pull request merged**